### ISSUE TITLE
Feature: Indexing fields by latitude/longitude rather than grid indices

### ISSUE THREAD

**Issue by michaeldenes** [OceanParcels/Parcels#1570]
A nice to have feature in `Parcels` would be the option to provide latitude and longitude indices, rather than the grid indices of the model.

For example, if I want to load a small region of a global ocean model in order to speed up the I/O of my simulation, I need to first identify the indices of the grid that line up with the latitude and longitude of said region (which can quite finicky to identify). I then pass these into my `FieldSet.from_nemo()` (or equivalent) function with the dictionary `indices = {'lat' : range(1000,2000), 'lon' : range(400,800)}`.

This becomes troublesome when using multiple fields using different grids, and so the grid indices will be different for different regions. It would be nice if I could instead pass a 'bounding box' or region in terms of latitude and longitude, rather than grid indices.

It would also be nice if this could deal with the periodic boundary, in the case where the region overlaps the periodic boundary of the model grid.

**Comment by erikvansebille** [comment]
Yes, I agree this would be a nice feature. And don't forget to also add it to the documentation/tutorials (see #1511)
